package.json

```diff
@@ -76,7 +76,7 @@
     "crosspath": "^1.0.0",
     "ts-node": "^10.8.0",
     "npm-check-updates": "^13.0.3",
-    "typescript": "^4.6.4",
+    "typescript": "~4.6.4",
```

Comment:
Note that this is just the _development_ dependency of TypeScript, and won't restrict library consumers from installing and using TypeScript 4.7. If your intention is to restrict the consumer facing version of TypeScript, it will need to be addressed below it in the package.json file under peer dependencies. However, I would prefer not to do that, and instead look into what the root cause of this apparent regression may be.

Reply:
I agree - my intent wasn't to fix the TS 4.7 issue, but to enable the tests to pass (and bring your attention to the 4.7 issue). 
 Glad to back this out - is that what you would prefer?

Reply:
Yes, I would prefer that we look into this TS 4.7 issue you mention here, maybe in a separate issue ☺️ I'm definitely going to be needing v4.7 support myself soon anyway, so we might as well leave it in.